lib/ts/TextBuffer.h

```diff
@@ -47,6 +47,15 @@ class TextBuffer
       copyFrom(rhs.bufPtr(), rhs.spaceUsed());
     }
   }
+  TextBuffer &
+  operator=(TextBuffer &&other)
+  {
+    if (this != &other) {
+      bufferStart       = other.bufferStart;
```

Comment:
If `bufferStart` is not `nullptr`, is it leaked?

Reply:
It was freed many times because an address in bufferStart was copied. With this change, only one instance will have a valid address for a buffer.

Reply:
and it will be freed when a TextBuffer that has the valid address is destruct.

Reply:
Consider this sequence -
```
TextBuffer a(100);
a = TextBuffer(200);
```
What happens to the 100 byte chunk of memory allocated in the first line during the execution of the second line?

Reply:
Ah, maybe I understand your point. Yeah, probably `this->bufferStart` need to be freed first.

Reply:
Added `ats_free()` before the substitution .